### PR TITLE
Add support for AOT, WVP, SCL, CLD, SNW data

### PR DESCRIPTION
**Pull request by an-ivanov**
SENTINEL2: Add support for exposing Level-2A AOT, WVP, SCL, CLD, SNW data in SAFE_COMPACT format

## What does this PR do?
Add support for exposing AOT, WVP, SCL, CLD, SNW data available in Level-2A products

## Tasklist
 - [ ] Add test case(s)
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

## Environment

Provide environment details, if relevant:

* OS: Windows 7 x64
* Compiler: VS2019

### CODE REVIEW

File: gdal/frmts/sentinel2/sentinel2dataset.cpp [hunk 1]
```diff
@@ -1559,13 +1571,19 @@ static CPLString SENTINEL2GetTilename(const CPLString& osGranulePath,
...
         else
         {
-            CPLDebug("SENTINEL2", "Invalid granule path: %s",
-                     osGranulePath.c_str());
+            osTile += "MSK_";
+            osTile += osBandName;
+            osTile += "PRB";
         }
-        osTile += osJPEG2000Name;
+        //else
```
Comment (rouault) [inline gdal/frmts/sentinel2/sentinel2dataset.cpp:1582]: please remove that commented code

File: gdal/frmts/sentinel2/sentinel2dataset.cpp [hunk 2]
```diff
@@ -2466,6 +2489,16 @@ GDALDataset *SENTINEL2Dataset::OpenL1C_L2A( const char* pszFilename,
                 osName = "0" + osName;
             oMapResolutionsToBands[psBandDesc->nResolution].insert(osName);
         }
+        if (eLevel == SENTINEL2_L2A )
+        {
+            for(unsigned int i = 0; i < NB_L2A_BANDS; ++i)
```
Comment (rouault) [inline gdal/frmts/sentinel2/sentinel2dataset.cpp:2494]: for( const auto& sL2ABandDesc: asL2ABandDesc)

File: gdal/frmts/sentinel2/sentinel2dataset.cpp [hunk 3]
```diff
@@ -3090,6 +3123,15 @@ GDALDataset *SENTINEL2Dataset::OpenL1C_L2ASubdataset( GDALOpenInfo * poOpenInfo,
...
```
Comment (rouault) [inline gdal/frmts/sentinel2/sentinel2dataset.cpp:3128]: for( const auto& sL2ABandDesc: asL2ABandDesc)

**Comment by rouault** [comment]
Unit tests  (autotest/gdrivers/sentinel2.py) should also be updated. See failures at https://travis-ci.com/github/OSGeo/gdal/jobs/319568924

File: gdal/frmts/sentinel2/sentinel2dataset.cpp (hunk 1 above)
Comment (an-ivanov) [reply gdal/frmts/sentinel2/sentinel2dataset.cpp]: done

**Comment by rouault** [comment]
There are still tests errors: https://travis-ci.com/github/OSGeo/gdal/jobs/319589593 . They might come from the fact that the tests products in autotest/ are fake datasets. Some "mandatory" files might be missing. If you need to add ones, you can copy an existing .jp2 file in those datasets (you'll notice that in fact they are not real JPEG2000 files but super small TIFF files which are good enough from the point of view of the GDAL driver)
To run locally the tests: pytest autotest/gdrivers/sentinel2.py

**Pull request merged**